### PR TITLE
Reduce the amount of API requests and improvements

### PR DESCRIPTION
**Pull request by maxaudron**
* add filter to ListAll*() function
    The filter allows us to, e.g. filter by name and dramaticly reduce the
    number of API requests as there is very little need to paginate anymore

* improve error messages
* remove unnecessary check for existence of resource  
    nutanix allows creating the same resource with the same name multiple
    times so we should allow this too.
    This also save us from doing some expensive API calls.

* get VM by ID for resourceNutanixVirtualMachineExists
    we have the ID available so it is not needed to loop through all the VMs

* remove ClusterIntentResource in favor of ClusterIntentResponse  
    ClusterIntentResource was only used in two places and is the exact same
    struct as ClusterIntentResponse.

### CODE REVIEW

**Comment by marinsalinas** [comment]
Hello @maxaudron does this close #87 ?

File: nutanix/resource_nutanix_image.go [hunk 1]
```diff
@@ -279,19 +279,10 @@ func resourceNutanixImageCreate(d *schema.ResourceData, meta interface{}) error
 	request.Metadata = metadata
 	request.Spec = spec
 
-	imageUUID, err := resourceNutanixImageExists(conn, n.(string))
```
Comment (marinsalinas) [inline nutanix/resource_nutanix_image.go:290]: Why did you remove this validation?

**Comment by coderGo93** [comment]
Hello, @maxaudron , there's a conflict, can you resolve it?

File: nutanix/resource_nutanix_image.go (hunk 1 above)
Comment (maxaudron) [reply nutanix/resource_nutanix_image.go:290]: The API does allow images with the same name so i don't see a reason why the provider should change that.

Should the API change to not allow this, the creation will fail anyways.

**Comment by PacoDw** [comment]
hello, @maxaudron could you do rebase to master branch plz to update the Travis file?

**Review by marinsalinas — APPROVED** [review]

**Review by marinsalinas — APPROVED** [review]
